File: InPlan.Web/Modules/Tenencia/Empresas/EmpresasRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Tenencia;

[ConnectionKey("Default"), Module("Tenencia"), TableName("Empresas")]
[DisplayName("Empresas"), InstanceName("Empresas")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed partial class EmpresasRow : Row<EmpresasRow.RowFields>, IIdRow, INameRow
{
    class RowTemplate
    {
        [DisplayName("Id Empresa"), Identity, IdProperty]
        public int? IdEmpresa { get; set; }

        [DisplayName("Empresa"), Size(200), NotNull, QuickSearch, NameProperty]
        public string DescripcionCorta { get; set; }

        [DisplayName("Descripción")]
        public string Descripcion { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Baja"), NotNull]
        public bool? Baja { get; set; }
    }
}

File: InPlan.Web/Modules/ServerTypes/Tenencia/EmpresasRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface EmpresasRow {
    IdEmpresa?: number;
    DescripcionCorta?: string;
    Descripcion?: string;
    Observaciones?: string;
    Baja?: boolean;
}

export abstract class EmpresasRow {
    static readonly idProperty = 'IdEmpresa';
    static readonly nameProperty = 'DescripcionCorta';
    static readonly localTextPrefix = 'Tenencia.Empresas';
    static readonly lookupKey = 'Tenencia.Empresas';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<EmpresasRow>('Tenencia.Empresas') }
    static async getLookupAsync() { return getLookupAsync<EmpresasRow>('Tenencia.Empresas') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<EmpresasRow>();
}